### PR TITLE
Fix/m policies

### PR DESCRIPTION
**Pull request by dgraeber**
*Issue #, if available:*
#445 

*Description of changes:*
Only try to delete the managed policy of no roles are attached

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: seedfarmer/commands/_stack_commands.py [hunk 1]
```diff
@@ -313,6 +327,14 @@ def deploy_module_stack(
 
     _logger.debug("project_managed_policy_output is : %s", stack_outputs)
     if project_managed_policy_stack_exists:
+        if stack_outputs.get("StackStatus") and "_IN_PROGRESS" in stack_outputs.get("StackStatus"):
+            _logger.info("The managed policy stack is not complete, waiting 60 seconds")
+            time.sleep(60)
```
Comment (srinivasreddych) [inline seedfarmer/commands/_stack_commands.py:332]: as discussed offline, its good to have this refactored with retry logic, while sleeping for 30 seconds

File: seedfarmer/commands/_stack_commands.py [hunk 2]
```diff
@@ -307,13 +321,29 @@ def deploy_module_stack(
...
+        project_managed_policy_stack_exists, stack_outputs = _check_stack_status()
+        if project_managed_policy_stack_exists:
+            if stack_outputs.get("StackStatus") and "_IN_PROGRESS" in stack_outputs.get("StackStatus"):
+                _logger.info("The managed policy stack is not complete, waiting 30 seconds")
+                time.sleep(30)
+                retries -= 1
+            else:
+                _logger.debug("project_managed_policy_output is : %s", stack_outputs)
+                project_managed_policy_arn = stack_outputs.get("ProjectPolicyARN", None)
+                retries = -1
```
Comment (srinivasreddych) [inline seedfarmer/commands/_stack_commands.py:343]: why should it retry once the `project_managed_policy_arn` is obtained? shoudn't this be `break` out of the while loop?

File: seedfarmer/commands/_stack_commands.py [hunk 3]
```diff
@@ -307,13 +321,29 @@ def deploy_module_stack(
...
+        else:
+            _logger.debug("project_managed_policy_output does not exist")
+            retries = -1
```
Comment (srinivasreddych) [inline seedfarmer/commands/_stack_commands.py:346]: Need for this retry?

we should retry only when the managed policy stack is in "*IN_PROGRESS" state

Comment (dgraeber) [reply seedfarmer/commands/_stack_commands.py:346]: this signals to break out of the loop...  sets retries equal to -1

File: seedfarmer/commands/_stack_commands.py (hunk 2 above)
Comment (dgraeber) [reply seedfarmer/commands/_stack_commands.py:343]: it is not...it is setting the retries value to -1

**Review by srinivasreddych — APPROVED** [review]

**Pull request merged**